### ISSUE TITLE
Failed to load node-pty module on synology DS218j

### ISSUE THREAD

**Issue by ivanroubicek** [oznu/homebridge-syno-spk#134]
### Raspberry Pi Model

Other (specify in description)

### Describe The Bug

I installed Homebridge 3.0.20 on my synology DS218j (DSM 7.1.1-42962 Update 5). I also performed all 3 optional steps. But when I open homebridge on port 8581, I am not able to connect (see attached logs for more information). 

I have tried `sudo hb-service rebuild`, which successfully executes `✔ Rebuilt homebridge-config-ui-x for Node.js v18.15.0.`, but nothing changes.

I see basically two problems in the logs

1. `Cannot find module '../build/Release/pty.node'`
2. `/lib/libc.so.6: version 'GLIBC_2.28' not found`

I looked to see what version of GLIBC I have on my synology and it is `/lib/libc.so.6 -> libc-2.26.so`.

### Logs

```text
[37m[11/30/2023, 10:36:16 AM][0m [36m[HB Supervisor][0m Creating default config.json: /volume1/homebridge/config.json
[37m[11/30/2023, 10:36:16 AM][0m [36m[HB Supervisor][0m OS: Linux 3.10.108 arm
[37m[11/30/2023, 10:36:16 AM][0m [36m[HB Supervisor][0m Node.js v18.15.0 /volume1/@appstore/Node.js_v18/usr/local/bin/node
[37m[11/30/2023, 10:36:16 AM][0m [36m[HB Supervisor][0m Homebridge Path: /volume1/homebridge/node_modules/homebridge/bin/homebridge
[37m[11/30/2023, 10:36:16 AM][0m [36m[HB Supervisor][0m UI Path: /var/packages/homebridge/target/app/lib/node_modules/homebridge-config-ui-x/dist/bin/standalone.js
prebuild-file-path: /volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/node_modules/@homebridge/node-pty-prebuilt-multiarch/prebuilds/linux-arm/node.abi108.node
innerError Error: Cannot find module '../build/Release/pty.node'
Require stack:
- /volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/node_modules/@homebridge/node-pty-prebuilt-multiarch/lib/prebuild-loader.js
- /volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/node_modules/@homebridge/node-pty-prebuilt-multiarch/lib/unixTerminal.js
- /volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/node_modules/@homebridge/node-pty-prebuilt-multiarch/lib/index.js
- /volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/dist/main.js
- /volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/dist/bin/hb-service.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/node_modules/@homebridge/node-pty-prebuilt-multiarch/lib/prebuild-loader.js:10:15)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)
    at Module.require (node:internal/modules/cjs/loader:1141:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/node_modules/@homebridge/node-pty-prebuilt-multiarch/lib/prebuild-loader.js',
    '/volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/node_modules/@homebridge/node-pty-prebuilt-multiarch/lib/unixTerminal.js',
    '/volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/node_modules/@homebridge/node-pty-prebuilt-multiarch/lib/index.js',
    '/volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/dist/main.js',
    '/volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/dist/bin/hb-service.js'
  ]
}
[0;37m[11/30/2023, 10:36:23 AM] [0m[0;36m[Homebridge UI][0m [0;31mError: /lib/libc.so.6: version `GLIBC_2.28' not found (required by /volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/node_modules/@homebridge/node-pty-prebuilt-multiarch/prebuilds/linux-arm/node.abi108.node)[0m
[0;37m[11/30/2023, 10:36:23 AM] [0m[0;36m[Homebridge UI][0m [0;31m[node-pty] Node.js v18.15.0[0m
[0;37m[11/30/2023, 10:36:23 AM] [0m[0;36m[Homebridge UI][0m [0;31m[node-pty] Failed to load node-pty module[0m
[0;37m[11/30/2023, 10:36:23 AM] [0m[0;36m[Homebridge UI][0m [0;31m[node-pty] This could be because the installation of this plugin did not complete successfully or you may have recently upgraded Node.js to a new major version.[0m
[0;37m[11/30/2023, 10:36:23 AM] [0m[0;36m[Homebridge UI][0m [0;31m[node-pty] Follow the steps below to resolve this issue.[0m
[0;37m[11/30/2023, 10:36:23 AM] [0m[0;36m[Homebridge UI][0m [0;33m[node-pty] Trying to rebuild automatically...[0m
[0;37m[11/30/2023, 10:36:23 AM] [0m[0;36m[Homebridge UI][0m [0;33m[node-pty] Path: /volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/node_modules/@homebridge/node-pty-prebuilt-multiarch[0m
[0;37m[11/30/2023, 10:37:05 AM] [0m[0;36m[Homebridge UI][0m [0;33m[node-pty] Module rebuilt successfully (maybe) - if you are still encountering errors follow the steps below.[0m
[0;37m[11/30/2023, 10:37:05 AM] [0m[0;36m[Homebridge UI][0m [0;33m[node-pty] From the terminal run this command to rebuild npm modules:
[0m
[0;37m[11/30/2023, 10:37:05 AM] [0m[0;36m[Homebridge UI][0m [0;33msudo hb-service rebuild
[0m
[37m[11/30/2023, 10:37:05 AM][0m [36m[HB Supervisor][0m ERROR: The user interface threw an unhandled error
Error: Node.js global modules rebuild required. See log errors above.
    at main (/volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/dist/main.js:31752:19)
    at 1808 (/volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/dist/main.js:31774:1)
    at __webpack_require__ (/volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/dist/main.js:32106:42)
    at /volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/dist/main.js:32134:1
    at /volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/dist/main.js:32240:3
    at Object.<anonymous> (/volume1/@appstore/homebridge/app/lib/node_modules/homebridge-config-ui-x/dist/main.js:32245:12)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)
```

**Comment by NorthernMan54** [comment]
Your DSM OS is pretty old, are their any updates for it ?

**Comment by ivanroubicek** [comment]
No.
`Status: Your DSM version is up-to-date.`

**Comment by ivanroubicek** [comment]
Just to be sure, I checked the synology website and although my NAS pretends to be up to date, I found a newer version 7.2.2 https://www.synology.com/en-global/support/download/DS218j?version=7.2#system. After updating, I reinstalled Homebridge just in case and everything seems to be working.

**Comment by DaGreenX** [comment]
Homebridge down again after hombebridge package synology update 3.0.2.1

**Comment by rh295** [comment]
same here :(

**Comment by NorthernMan54** [comment]
Tracking here - #141

**Comment by DaGreenX** [comment]
trying this https://github.com/homebridge/homebridge-syno-spk/issues/134 
updating DSM 7.1.1-42962  to 7.2.1-69057 Update 3

what a surprise! 
Reboot after update, everything's fucked! 7 packages in error, including file station! Very nice!

**Comment by DaGreenX** [comment]
what a surprise! 
Reboot after update, everything's fucked! 7 packages in error, including file station! Very nice!

**Comment by rbyrbt** [comment]
> Just to be sure, I checked the synology website and although my NAS pretends to be up to date, I found a newer version 7.2.2 https://www.synology.com/en-global/support/download/DS218j?version=7.2#system. After updating, I reinstalled Homebridge just in case and everything seems to be working.

I had the same issue DS918+ was stuck on DSM 7.1.1 (claiming it was up-to-date). Manually updated to DSM 7.2.1, Node 20 (available in Package Center on DSM 7.2), and Homebridge 3.1.1 (Package Center) and everything is appears to be running smoothly.

**Comment by rbyrbt** [comment]
> trying this #134 updating DSM 7.1.1-42962 to 7.2.1-69057 Update 3
> 
> what a surprise! Reboot after update, everything's fucked! 7 packages in error, including file station! Very nice!

Make sure you are using this link to get the correct DSM download(s) for your NAS: https://www.synology.com/en-global/support/download After selecting my NAS, I used the option below where you select the upgrade path eg: DSM 7.1.1 to DSM 7.2.1 to get all the downloads.

Basically something got messed up with the DSM auto-updater. More here: https://www.synology.com/en-us/releaseNote/DSM
